### ISSUE TITLE
Weight decay in AdamW

### ISSUE THREAD

**Issue by jeongukjae** [pytorch/pytorch#38853]
## 🐛 Bug

As the original paper(https://arxiv.org/pdf/1711.05101.pdf, green boxes) shows 

<img width="604" alt="Screen Shot 2020-05-21 at 12 40 39 PM" src="https://user-images.githubusercontent.com/8815362/82520936-49782e00-9b60-11ea-869e-1d9e06bbea26.png">

the formula of applying weight decay to Adam should be 
`\theta_t = (1 - \lambda) \theta_{t - 1}` or
`\theta_t = (1 - {schedule multiplier} * \lambda) \theta_{t - 1}`.

But, AdamW implementation in master branch (https://github.com/pytorch/pytorch/blob/master/torch/optim/adamw.py#L73) applies weight decay with learning rate.

`\theta_t = (1 - \lambda * {learning rate}) \theta_{t - 1}`

## Expected behavior

I think this line(https://github.com/pytorch/pytorch/blob/a8d8fc553229731c2ca491fefe18ff977c7e8af0/torch/optim/adamw.py#L73) should be `p.mul_(1 - group['weight_decay'])`.

**Comment by yuhaozhang** [comment]
Any updates on this issue? I recently discovered this issue after I observed some very different behaviors when applying a similar weight decay term to Adam and AdamW. I think this implementation difference with the original AdamW paper should at least be mentioned in the documentation of AdamW?

**Comment by vincentqb** [comment]
I believe this was also discussed [here](https://github.com/pytorch/pytorch/pull/21250#issuecomment-520559993). Is that correct?

Yes, I agree it would be a good idea to detail this in the documentation. Do you want to open a pull request to update the documentation?

**Comment by jeongukjae** [comment]
@vincentqb Yes, discussion at that link is same with this issue. I can open a pull request soon.
